neuraxle/metaopt/auto_ml.py

```diff
@@ -0,0 +1,358 @@
+"""
+Neuraxle's Automatic Machine Learning Classes
+====================================
+All steps, and abstractions needed to build Automatic Machine Learning algorithms in Neuraxle.
+
+..
+    Copyright 2019, Neuraxio Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+"""
+import copy
+import glob
+import hashlib
+import json
+import os
+from abc import ABC, abstractmethod
+from typing import Tuple, List
+
+from neuraxle.base import MetaStepMixin, BaseStep, NonTransformableMixin
+from neuraxle.hyperparams.space import HyperparameterSamples, HyperparameterSpace
+from neuraxle.metaopt.random import BaseCrossValidationWrapper, KFoldCrossValidationWrapper
+
+
+class HyperparamsRepository(ABC):
+    """
+    Hyperparams repository that saves hyperparams, and scores for every AutoML trial.
+
+    .. seealso::
+        :class:`HyperparamsRepository`,
+        :class:`HyperparameterSamples`,
+        :class:`AutoMLSequentialWrapper`
+    """
+
+    @abstractmethod
+    def create_new_trial(self, hyperparams: HyperparameterSamples):
+        """
+        Create new hyperperams trial.
+
+        :param hyperparams: hyperparams
+        :type hyperparams: HyperparameterSamples
+        :return:
+        """
+        pass
+
+    @abstractmethod
+    def load_all_trials(self) -> Tuple[List[HyperparameterSamples], List[float]]:
+        """
+        Load all hyperparameter trials with their corresponding score.
+
+        :return: (hyperparams, scores)
+        """
+        pass
+
+    @abstractmethod
+    def save_score_for_trial(self, hyperparams: HyperparameterSamples, score: float):
+        """
+        Save hyperparams, and score for a trial.
+
+        :return: (hyperparams, scores)
+        """
+        pass
+
+    def get_best_hyperparams(self, higher_score_is_better=True) -> HyperparameterSamples:
```

Comment:
`higher_score_is_better=True` should probably be somewhere else. This might be a symptom of something else: I feel like this method shouldn't belong in the repository, that might be the reason. It might belong into the AutoMLWrapper or some other related class.

Reply:
created a method get_hyperparams_from_trials inside AutoMLStrategy